synapse/federation/federation_client.py

```diff
@@ -264,6 +264,62 @@ async def backfill(
 
         return pdus
 
+    async def get_pdu_from_destination_raw(
```

Comment:
This was used in a previous commit from this PR where I wanted to see if the insertion event existed on the federating homeserver asking about backfill. **But now this function is no longer used**. It seems like a nice refactor of some of the logic out. And the caching/rate-limiting is handled in the `get_pdu` consumer. Seems like it may be worth keeping but feel free to poke to revert.

https://github.com/matrix-org/synapse/blob/e0ff66dfd9949d83d9be13f62faf96b68446065e/synapse/handlers/federation.py#L1129-L1147

Reply:
Yeah, I think this PR is a bit unwieldy as it is. Can we split out the refactors and empty prev events please? Just to get the diff down to the actual change :slightly_smiling_face:

Reply:
Split out in https://github.com/matrix-org/synapse/pull/11242